Sync/BatchingClient.swift

```diff
@@ -97,7 +115,28 @@ open class Sync15BatchClient<T: CleartextPayloadJSON> {
             >>> succeed
     }
 
-    open func addRecords(_ records: [Record<T>]) -> Success {
+    // If in batch mode, will discard the batch if any record fails
+    open func endSingleBatch() -> Success {
```

Comment:
I think you want to return `DeferredResponse` here, which will give you the timestamp you need.